runtime/runtime.go

```diff
@@ -349,6 +362,123 @@ func (r *interpreterRuntime) newAuthAccountValue(
 	)
 }
 
+func (r *interpreterRuntime) InvokeContractFunction(
+	contractLocation common.AddressLocation,
+	functionName string,
+	arguments []interpreter.Value,
+	argumentTypes []sema.Type,
+	context Context,
+) (cadence.Value, error) {
+	context.InitializeCodesAndPrograms()
+
+	runtimeStorage := newRuntimeStorage(context.Interface)
+
+	var interpreterOptions []interpreter.Option
+	var checkerOptions []sema.Option
+
+	functions := r.standardLibraryFunctions(
+		context,
+		runtimeStorage,
+		interpreterOptions,
+		checkerOptions,
+	)
+
+	// create interpreter
+	_, inter, err := r.interpret(
+		nil,
+		context,
+		runtimeStorage,
+		functions,
+		stdlib.BuiltinValues,
+		interpreterOptions,
+		checkerOptions,
+		nil,
+	)
+	if err != nil {
+		return nil, newError(err, context)
+	}
+
+	// ensure the contract is loaded
+	inter = inter.EnsureLoaded(contractLocation)
+
+
+	for i, argumentType := range argumentTypes {
+		arguments[i] = r.convertArgument(
+			arguments[i],
+			argumentType,
+			context,
+			runtimeStorage,
+			interpreterOptions,
+			checkerOptions,
+		)
+	}
+
+	contractValue, err := inter.GetContractComposite(contractLocation)
+	if err != nil {
+		return nil, newError(err, context)
+	}
+
+	// prepare invocation
+	invocation := interpreter.Invocation{
+		Self:               contractValue,
+		Arguments:          arguments,
+		ArgumentTypes:      argumentTypes,
+		TypeParameterTypes: nil,
+		GetLocationRange: func() interpreter.LocationRange {
+			return interpreter.LocationRange{
+				Location: context.Location,
+			}
+		},
+		Interpreter: inter,
+	}
+
+	contractMember := contractValue.GetMember(inter, invocation.GetLocationRange, functionName)
+
+	contractFunction, ok := contractMember.(interpreter.FunctionValue)
+	if !ok {
+		return nil, newError(
+			interpreter.NotInvokableError{
+				Value: contractFunction,
+			},
+			context)
+	}
+
+	value, err := inter.InvokeFunction(contractFunction, invocation)
+	if err != nil {
+		return nil, newError(err, context)
+	}
+
+
+	// Write back all stored values, which were actually just cached, back into storage
+	runtimeStorage.writeCached(inter)
```

Comment:
@turbolent added this line otherwise changes were not persisted no the fvm side

Reply:
@janezpodhostnik looks good! 👍